packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroupOption/ChoiceGroupOption.base.tsx

```diff
@@ -1,22 +1,44 @@
 import * as React from 'react';
 import { Image } from '../../Image';
 import { Icon } from '../../Icon';
+import { TooltipHost, TooltipOverflowMode } from '../../Tooltip';
 import { IClassNames } from '@uifabric/utilities/lib/IClassNames';
 import {
   IChoiceGroupOptionProps,
   IChoiceGroupOptionStyleProps,
   IChoiceGroupOptionStyles
 } from './ChoiceGroupOption.types';
-import { BaseComponent, classNamesFunction, getNativeProps, inputProperties, createRef } from '../../../Utilities';
+import {
+  BaseComponent,
+  classNamesFunction,
+  getNativeProps,
+  inputProperties,
+  createRef,
+  hasOverflow
+} from '../../../Utilities';
 
 const getClassNames = classNamesFunction<IChoiceGroupOptionStyleProps, IChoiceGroupOptionStyles>();
+const defaultImageSize = 32;
+
+export interface IChoiceGroupOptionState {
+  hasOverflow?: boolean;
+}
 
-export class ChoiceGroupOptionBase extends BaseComponent<IChoiceGroupOptionProps, any> {
+export class ChoiceGroupOptionBase extends BaseComponent<IChoiceGroupOptionProps, IChoiceGroupOptionState> {
   private _inputElement = createRef<HTMLInputElement>();
   private _classNames: IClassNames<IChoiceGroupOptionStyles>;
+  private _labelWrapperElementList: HTMLDivElement[] = [];
 
   constructor(props: IChoiceGroupOptionProps) {
     super(props);
+
+    this.state = {
+      hasOverflow: false
+    };
+  }
+
+  public componentDidMount(): void {
+    this._calculateLabelTextOverflow();
```

Comment:
This will cause a synchronous reflow! You shouldn't access hasOverflow without being in a RAF.

Reply:
I wonder if there is some way we can make this way more obvious. Like name "hasOverflow" to "asyncHasOverflow" and do the raf in there.